### PR TITLE
fix: percentage calculation per total batches array (MAPCO-2984)

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                      |

* Resolve the problem that each batch array the progress percentage was overrided per batch

* making the calcultion of percentage to total amount of tiles as sum of all batches

### CODE REVIEW

File: MergerService/src/run.cs [hunk 1]
```diff
@@ -270,24 +275,24 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils, string? managerCallba
                         stopWatch.Reset();
                         stopWatch.Start();
 
-                        int totalTileCount = (int)bounds.Size();
+                        int singleTileBatchCount = (int)bounds.Size();
```
Comment (asafMasa) [inline MergerService/src/run.cs:278]: check with @shimoncohen what is the size of tile in zoom 0 on zoom 21, is it long or int?

**Review by asafMasa — APPROVED** [review]

File: MergerService/Utils/TaskUtils.cs [hunk 2]
```diff
@@ -148,6 +148,7 @@ public void UpdateReject(string jobId, string taskId, int attempts, string reaso
             }
         }
 
+
```
Comment (asafMasa) [inline MergerService/Utils/TaskUtils.cs:151]: remove irrelevant space

File: MergerService/appsettings.json [hunk 3]
```diff
@@ -1,22 +1,26 @@
 {
   "GENERAL": {
     "validate": false,
-    "inputPath": "input",
-    "gpkgPath": "gpkg",
-    "filePath": "tiles"
+    "inputPath": "/home/ronenk1/Documents/input",
```
Comment (asafMasa) [inline MergerService/appsettings.json:4]: revert

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: MergerService/Utils/TaskUtils.cs (hunk 2 above)
Comment (ronenkapelian) [reply MergerService/Utils/TaskUtils.cs:151]: done

File: MergerService/appsettings.json (hunk 3 above)
Comment (ronenkapelian) [reply MergerService/appsettings.json:4]: done

**Review by asafMasa — APPROVED** [review]

File: MergerService/src/run.cs [hunk 4]
```diff
@@ -270,24 +277,25 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils, string? managerCallba
...
-                        List<IData> sources = this.BuildDataList(metadata.Sources, totalTileCount);
+                        List<IData> sources = this.BuildDataList(metadata.Sources, this._batchSize);
                         IData target = sources[0];
 
-                        List<Tile> tiles = new List<Tile>(totalTileCount);
+                        // TODO: fix to use inner batch size (add iteration inside loop below)
+                        List<Tile> tiles = new List<Tile>((int)singleTileBatchCount);
 
-                        this._logger.LogInformation($"Total amount of tiles to merge: {totalTileCount}");
+                        this._logger.LogInformation($"Total amount of tiles to merge for current batch iteration: {singleTileBatchCount}");
```
Comment (shimoncohen) [inline MergerService/src/run.cs:298]: Remove the word `iteration`.

File: MergerService/src/run.cs [hunk 5]
```diff
@@ -439,5 +435,27 @@ private bool Validate(IData target, TileBounds bounds)
 
             return hasSameTiles;
         }
+
+        public void UpdateRelativeProgress(MergeTask task, long currentAmount, long totalAmount, ITaskUtils taskUtils)
```
Comment (shimoncohen) [inline MergerService/src/run.cs:439]: Didn't we say this should be part of `Task`?

**Review by shimoncohen — CHANGES_REQUESTED** [review]

File: MergerService/src/run.cs [hunk 6]
```diff
@@ -260,6 +263,10 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils, string? managerCallba
                 //taskActivity.AddTag("jobId", task.jodId);
                 //taskActivity.AddTag("taskId", task.id);
 
+                long totalTileCount = metadata.Batches.Sum(batch => batch.Size());
+                long overallTileProgressCount = 0;
+                this._logger.LogInformation($"Total amount of tiles to merge [All external batches]: {totalTileCount}");
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply MergerService/src/run.cs:268]: done

File: MergerService/src/run.cs (hunk 4 above)
Comment (ronenkapelian) [reply MergerService/src/run.cs:298]: done

File: MergerService/src/run.cs (hunk 5 above)
Comment (ronenkapelian) [reply MergerService/src/run.cs:439]: i decide it more relevant as some external logic of run.cs scope

**Pull request merged**